cle/backends/java/apk.py

```diff
@@ -73,6 +76,79 @@ def __init__(self, apk_path, binary_stream, entry_point=None, entry_point_params
                                   jni_libs_ld_path=jni_libs_ld_path,
                                   **options)
 
+        # the lifecycle needs to support of pyaxmlparser
+        if PYAXMLPARSER_INSTALLED:
+            self.components = {'activity': [], 'service': [], 'receiver': [], 'provider': []}
+            self.callbacks = {'activity': [], 'service': [], 'receiver': [], 'provider': []}
+            self._set_lifecycle(apk_parser)
+
+    def _set_lifecycle(self, apk_parser):
+        """
+        Set components with callbacks of APK lifecycle.
+
+        :param pyaxmlparser apk_parser: XML Parser of the APK.
+        """
+
+        component_getter = {'activity': apk_parser.get_activities,
+                            'service': apk_parser.get_services,
+                            'receiver': apk_parser.get_receivers,
+                            'provider': apk_parser.get_providers}
+
+        for key, getter in component_getter.items():
+            class_names = getter()
+            self.components[key], self.callbacks[key] = self._extract_lifecycle(class_names, key)
+
+    def _extract_lifecycle(self, class_name, component_kind):
+        """
+        Extract components with callbacks from class names and component kind.
+        Use general callback name for each component by component kind
+
+        :param str cls:             Name of the class.
+        :param str component_kind:  Kind of the component. (activity, service, receiver, provider)
+        :return list components:    The list of class objects which are components.
+        :return list callbacks:     The list of method objects which are callbacks.
```

Comment:
Please use inline type annotations.

Reply:
I altered to inline type annotation. Thanks for your feedbacks!